pkg/util/template/resource.go

```diff
@@ -25,34 +26,60 @@ type HTML struct {
 	Name string
 	// ComponentDependencies is the HTML component templates this template depends on.
 	ComponentDependencies []*HTML
-	// IsFindAllowedInFs returns an boolean indicating if the Fs can be used to find this resources
-	// For example, if you don't want a html inside the app fs to be used, it should return false.
-	IsFindAllowedInFs FindResourcesFsFilter
+}
+
+type MessageHTML struct {
+	// Name is the name of template
+	Name string
 }
 
 var _ resource.Descriptor = &HTML{}
+var _ resource.Descriptor = &MessageHTML{}
 
-func (t *HTML) templateResource() {}
+func (t *HTML) templateResource()        {}
+func (t *MessageHTML) templateResource() {}
 
 func (t *HTML) MatchResource(path string) (*resource.Match, bool) {
+	if strings.HasPrefix(t.Name, "messages/") {
+		return nil, false
+	}
+	return matchTemplatePath(path, t.Name)
+}
+
+func (t *MessageHTML) MatchResource(path string) (*resource.Match, bool) {
+	if !strings.HasPrefix(t.Name, "messages/") {
+		return nil, false
+	}
```

Comment:
No need to add this condition. We assume the calls to RegisterMessageHTML are correct, that is, no one will ever use RegisterMessageHTML to register regular HTML.

Reply:
removed